src/Scripting/Core/ScriptOptions.cs

```diff
@@ -92,6 +93,21 @@ private static ImmutableArray<MetadataReference> GetDefaultMetadataReferences()
         public ImmutableArray<string> Imports { get; private set; }
 
         /// <summary>
+        /// Specifies whether debugging symbols should be emitted.
+        /// </summary>
+        public bool EmitDebugInformation => DebugInformationFormat != null;
+
+        /// <summary>
+        /// <see cref="DebugInformationFormat"/> to be used for debugging symbols if needed
+        /// </summary>
+        public DebugInformationFormat? DebugInformationFormat { get; private set; } = null;
+
+        /// <summary>
+        /// Specifies the encoding to be used for the code.
+        /// </summary>
+        public Encoding Encoding { get; private set; } = Encoding.UTF8;
```

Comment:
>Encoding [](start = 24, length = 8)

I'd call it FileEncoding and explained in the comment that this is only relevant for debugging scripts that were read from a file or they will be saved to a file for debugging purposes.